From 89dbb1cba76f12db000f5052cd14b43fbd131be7 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Wed, 27 Jan 2016 13:31:13 -0800
Subject: [PATCH 060/194] Snap: Don't crash when hardcoded gallery intent fails

* startActivity may also throw IllegalArgumentException in addition to
  ActivityNotFoundException when the component can't be found.

Change-Id: Ia3124591c625f4c249f3c6f76a226e8178df6e9d
---
 src/com/android/camera/CameraActivity.java | 24 ++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/src/com/android/camera/CameraActivity.java b/src/com/android/camera/CameraActivity.java
index bc43aa178..2e7ecea2b 100755
--- a/src/com/android/camera/CameraActivity.java
+++ b/src/com/android/camera/CameraActivity.java
@@ -616,14 +616,22 @@ public class CameraActivity extends Activity
             intent.putExtra(KEY_TOTAL_NUMBER, (adapter.getTotalNumber() -1));
             startActivity(intent);
         } catch (ActivityNotFoundException ex) {
-            try {
-                Log.w(TAG, "Gallery not found");
-                Intent intent = new Intent(Intent.ACTION_VIEW, uri);
-                intent.putExtra(KEY_FROM_SNAPCAM, true);
-                startActivity(intent);
-            } catch (ActivityNotFoundException e) {
-                Log.w(TAG, "No Activity could be found to open image or video");
-            }
+            gotoViewPhoto(uri);
+        } catch (IllegalArgumentException ex) {
+            gotoViewPhoto(uri);
+        }
+    }
+
+    private void gotoViewPhoto(Uri uri) {
+        try {
+            Log.w(TAG, "Gallery not found");
+            Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+            intent.putExtra(KEY_FROM_SNAPCAM, true);
+            startActivity(intent);
+        } catch (ActivityNotFoundException e) {
+            Log.w(TAG, "No Activity could be found to open image or video");
+        } catch (IllegalArgumentException e) {
+            Log.w(TAG, "No Activity could be found to open image or video");
         }
     }
 
-- 
2.17.1

